manim/scene/scene_file_writer.py

```diff
@@ -483,17 +483,16 @@ def write_frame(
             )
 
     def output_image(
-        self, image: Image.Image, target_dir: StrPath, ext: str, zero_pad: bool
+        self, image: Image.Image, target_dir: StrPath, ext: str, zero_pad: int
     ) -> None:
         if zero_pad:
             image.save(f"{target_dir}{str(self.frame_count).zfill(zero_pad)}{ext}")
         else:
             image.save(f"{target_dir}{self.frame_count}{ext}")
         self.frame_count += 1
 
-    def save_final_image(self, image: Image.Image) -> None:
-        """The name is a misnomer. This method saves the image
-        passed to it as an in the default image directory.
+    def save_image(self, image: PixelArray) -> None:
```

Comment:
`image` should be a `PIL.Image.Image`, because `PixelArray` is a NumPy array and those don't have a `save` method (there is a `numpy.save()` function, but it's used differently). Around the source code, the argument which is passed to this method is effectively a `PIL.Image Image`.
```suggestion
    def save_image(self, image: Image.Image) -> None:
```